src/specific/file.c

```diff
@@ -575,19 +576,18 @@ static bool LoadSamples(MYFILE *fp)
         return false;
     }
 
-    int32_t *sample_offsets =
-        Memory_Alloc(sizeof(int32_t) * num_samples, MEM_BUF_SAMPLE_OFFSETS);
+    int32_t *sample_offsets = S_Memory_Alloc(sizeof(int32_t) * num_samples);
     FileRead(sample_offsets, sizeof(int32_t), num_samples, fp);
 
-    char **sample_pointers =
-        Memory_Alloc(sizeof(char *) * num_samples, MEM_BUF_SAMPLE_OFFSETS);
+    char **sample_pointers = S_Memory_Alloc(sizeof(char *) * num_samples);
```

Comment:
I feel this is now overkill as we have c99 and this only exists in this scope as they are removed bellow. We could just do `int32 sample_offsets[num_samples];` and likewise for the pointers and just let them fall out of scope. The number of samples is probably below 256 to which adding 2K onto the stack is nothing these days.